test/integration/supplemental_groups_test.go

```diff
@@ -0,0 +1,108 @@
+// +build integration
+
+package integration
+
+import (
+	"strconv"
+	"testing"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+
+	"github.com/fsouza/go-dockerclient"
+	"github.com/openshift/origin/pkg/cmd/server/bootstrappolicy"
+	testutil "github.com/openshift/origin/test/util"
+	testserver "github.com/openshift/origin/test/util/server"
+	"k8s.io/kubernetes/pkg/securitycontext"
+)
+
+// TestSupplementalGroups tests that a requested fs group and supplemental group
+// pass through the system and to the resulting docker config.
+func TestSupplementalGroups(t *testing.T) {
+	kubeClient := setupSupplementalGroupsTest(t)
+
+	fsGroup := int64(1111)
+	supGroup := int64(2222)
+
+	pod := &kapi.Pod{
+		ObjectMeta: kapi.ObjectMeta{
+			Name: "supplemental-groups",
+		},
+		Spec: kapi.PodSpec{
+			SecurityContext: &kapi.PodSecurityContext{
+				FSGroup:            &fsGroup,
+				SupplementalGroups: []int64{supGroup},
+			},
+			Containers: []kapi.Container{
+				{
+					Name:  "supplemental-groups",
+					Image: "openshift/origin-pod",
+				},
+			},
+		},
+	}
+
+	// create the pod to allow it to go through the admission control which should fail if the
+	// user submitting does not have access to request specific groups.
+	_, err := kubeClient.Pods(testutil.Namespace()).Create(pod)
+	if err != nil {
+		t.Fatalf("unable to create pod: %v", err)
+	}
+	defer kubeClient.Pods(testutil.Namespace()).Delete(pod.Name, nil)
+
+	// get the pod which should now be updated.  If admission mutated the pod unexpectedly
+	// then the checks for our submitted groups will fail (below)
+	updatedPod, err := kubeClient.Pods(testutil.Namespace()).Get(pod.Name)
+	if err != nil {
+		t.Fatalf("unable to get updated pod: %v", err)
+	}
+
+	// simulate what the kubelet would do to modify the container's host config
```

Comment:
if tmpfs for the SA token is the issue, you can create a volume mount at the expected SA token mount path and the SA token won't get added.